sre/storage/kvStorage_test.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPutandGet(t *testing.T) {
+
+	m := Map{}
+	tables := []struct {
+		x string
+		y interface{}
+	}{
+		{"a", 1},
+		{"b", 2.0},
+		{"c", "harry"},
+	}
+	for _, table := range tables {
+		m.Put(table.x, table.y)
+		val, _ := m.Get(table.x)
+		// assert.EqualError(t, err, nil)
+		assert.Equal(t, val, table.y)
+	}
+
+}
+func TestDelete(t *testing.T) {
+	m := Map{}
+	tables := []struct {
```

Comment:
ideally, you can have only one value. why have a struct? 

simply do this.
```
m.Put("a", 1}
m.Delete("a")
``` and so on

Reply:
so here i am checking the interface design of map too like inseting different type of values